active/0002-centalized-mgmt-resource-types-provider-versions.md

```diff
@@ -0,0 +1,258 @@
+
+## Terms and definitions
+
+- **Resource Types Provider Package aka provider package** - The OCI Registry artifact that contains the resource type definitions used by Bicep to identify the resource types properties, functions and other relevant type information that it uses to generate ARM-JSON.
+- **types.json protocol** - It is a serialization protocol used by Bicep to encode type information as data that can be loaded to the Bicep runtime.
+
+## Motivation
+
+The [resource type providers dynamic loading feature](https://github.com/Azure/bicep/issues/5453) introduces a change in provider declaration statement syntax to include the registry address and version that Bicep will use to resolve resource type declarations.
+
+**Provider Declaration Syntax**
+
+```bicep
+provider 'br:mcr.microsoft.com/bicep/providers/az@1.2.3' // fully-qualified
+provider 'br/public:az@1.2.3' // uses provider aliasing defined in bicepconfig.json
+```
+
+We also currently support the following syntax for "built-in" providers:
+
+```bicep
+provider 'sys@1.0.0'
+```
+
+In the above cases, the provider can be referenced by using its symbol which is derived from the segment to the left of the '@' sign. (`az` and `sys` for the declarations above)
+
+Both syntaxes also support aliasing of that symbol reference:
+
+```bicep
+provider 'br/public:az@0.2.3' as myAz
+provider 'sys@1.0.0' as mySys
+```
+
+The current syntax (shown above) presents several challenges:
+
+1. The new syntax is quite verbose, and cannot be controlled centrally. For users with a large set of files, picking up the latest version of `az` will require modifying every single file.
+1. The aliasing (assuming that the last segment of the registry namespace == the provider name) & versioning (having an `@` character instead of a `:` which is familiar to module users) behavior of the new syntax can be confusing.
+1. Its deceiving and confusing to allow `sys` to have a pinned version fixed to `1.0.0` - this **should** imply that there are no changes between Bicep versions, which isn't realistic.
+
+- It's an "either/or" - either users follow the current behavior (`az` imported by default) **or** they're using the new syntax. We don't have a plan for how to combine the two.
+- With [Radius](https://github.com/radius-project/radius), it's conceivable that we may want users to **not** have `az` imported by default.
+
+### Proposed Changes
+
+- [Resolution of namespace identifiers by VSCode extension when inspecting published module sources](#resolution-of-namespace-identifiers-by-vscode-extension-when-inspecting-published-module-sources)
+- [Wildcard semantics for provider versions in `bicepconfig.json`](#wildcard-semantics-for-provider-versions-in-bicepconfigjson)
```

Comment:
Both of these link to headings under "out of scope". Are they out of scope or in scope?

Reply:
things got scrambled here, fixing now...